workflow/controller/controller.go

```diff
@@ -484,6 +484,9 @@ func (wfc *WorkflowController) processNextItem() bool {
 		return true
 	}
 
+	// this will ensure we process every incomplete workflow once every 20m
+	wfc.wfQueue.AddAfter(key, workflowResyncPeriod)
```

Comment:
can we add the workflow on end of process Instead of beginning?  In this case, we can avoid adding completed workflows. Will this add wf with 20m for every processing?

Reply:
I wil update to use a defer

Reply:
Actually, I don't want to do that. This is hard to test code and I don't want to make it at all complex. If it is complete, then next time it will just be ignored. I want to leave it like this.

Reply:
I think it should be fine. I tested `AddAfter` is dedup item and process the earliest one if it is added multiple times with different durations.  Only side effect, all completed workflows will stay in queue another 20 mins.